### PR TITLE
Port System.Runtime.Intrinsics new docs

### PR DESCRIPTION
**Pull request by carlossanlop**
@dotnet/area-system-runtime-intrinsics @directhex @gewarren PTAL

### CODE REVIEW

File: xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_cvtsi64_sd (__m128d a, __int64 b) VCVTUSI2SD xmm1, xmm2, r/m64 This intrinsic is only available on 64-bit processes</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml:43]: Should "This intrinsic is only available on 64-bit processes" be in the remarks instead?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>To be added.</summary>
+    <summary>This class provides access to X86 AVX10.1 hardware instructions via intrinsics</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:23]:
```suggestion
    <summary>Provides access to X86 AVX10.1 hardware instructions via intrinsics.</summary>
```

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 3]
```diff
@@ -6577,7 +6577,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_getexp_sd (__m128d a, __m128d b) VGETEXPSD xmm1 {k1}{z}, xmm2, xmm3/m64{sae} The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:6580]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 4]
```diff
@@ -6604,7 +6604,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128 _mm_getexp_ss (__m128 a, __m128 b) VGETEXPSS xmm1 {k1}{z}, xmm2, xmm3/m32{sae} The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:6607]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 5]
```diff
@@ -6844,7 +6844,7 @@
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_getmant_sd (__m128d a, __m128d b) VGETMANTSD xmm1 {k1}{z}, xmm2, xmm3/m64{sae} The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:6847]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 6]
```diff
@@ -6880,7 +6880,7 @@
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128 _mm_getmant_ss (__m128 a, __m128 b) VGETMANTSS xmm1 {k1}{z}, xmm2, xmm3/m32{sae} The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:6883]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 7]
```diff
@@ -8875,7 +8875,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_rcp14_sd (__m128d a, __m128d b) VRCP14SD xmm1 {k1}{z}, xmm2, xmm3/m64 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:8878]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 8]
```diff
@@ -8902,7 +8902,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128 _mm_rcp14_ss (__m128 a, __m128 b) VRCP14SS xmm1 {k1}{z}, xmm2, xmm3/m32 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:8905]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 9]
```diff
@@ -9079,7 +9079,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_rsqrt14_sd (__m128d a, __m128d b) VRSQRT14SD xmm1 {k1}{z}, xmm2, xmm3/m64 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:9082]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 10]
```diff
@@ -9106,7 +9106,7 @@
       <Docs>
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128 _mm_rsqrt14_ss (__m128 a, __m128 b) VRSQRT14SS xmm1 {k1}{z}, xmm2, xmm3/m32 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:9109]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 11]
```diff
@@ -9346,7 +9346,7 @@
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_reduce_sd(__m128d a, __m128d b, int imm); VREDUCESD xmm1 {k1}{z}, xmm2, xmm3/m64{sae}, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:9349]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 12]
```diff
@@ -9382,7 +9382,7 @@
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128 _mm_reduce_ss(__m128 a, __m128 b, int imm); VREDUCESS xmm1 {k1}{z}, xmm2, xmm3/m32{sae}, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:9385]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 13]
```diff
@@ -10598,7 +10598,7 @@
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_roundscale_sd (__m128d a, __m128d b, int imm) VRNDSCALESD xmm1 {k1}{z}, xmm2, xmm3/m64{sae}, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:10601]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 14]
```diff
@@ -10634,7 +10634,7 @@
         <param name="upper">To be added.</param>
         <param name="value">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128 _mm_roundscale_ss (__m128 a, __m128 b, int imm) VRNDSCALESS xmm1 {k1}{z}, xmm2, xmm3/m32{sae}, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other scalar APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:10637]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 15]
```diff
@@ -11784,7 +11784,7 @@
         <param name="b">To be added.</param>
         <param name="c">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128i _mm_ternarylogic_si128 (__m128i a, __m128i b, __m128i c, byte imm) VPTERNLOGD xmm1 {k1}{z}, xmm2, xmm3/m128, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other bitwise APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:11787]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 16]
```diff
@@ -11822,7 +11822,7 @@
         <param name="b">To be added.</param>
         <param name="c">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128d _mm_ternarylogic_pd (__m128d a, __m128d b, __m128d c, int imm) VPTERNLOGQ xmm1 {k1}{z}, xmm2, xmm3/m128/m64bcst, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other bitwise APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:11825]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 17]
```diff
@@ -11860,7 +11860,7 @@
         <param name="b">To be added.</param>
         <param name="c">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128i _mm_ternarylogic_si128 (__m128i a, __m128i b, __m128i c, short imm) VPTERNLOGD xmm1 {k1}{z}, xmm2, xmm3/m128, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other bitwise APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:11863]: Should some of this go in remarks?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 18]
```diff
@@ -11974,7 +11974,7 @@
         <param name="b">To be added.</param>
         <param name="c">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m128i _mm_ternarylogic_si128 (__m128i a, __m128i b, __m128i c, byte imm) VPTERNLOGD xmm1 {k1}{z}, xmm2, xmm3/m128, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other bitwise APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:11977]: Should some of this go in remarks? (Same for next two changes)

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 19]
```diff
@@ -12164,7 +12164,7 @@
         <param name="b">To be added.</param>
         <param name="c">To be added.</param>
         <param name="control">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>__m256i _mm256_ternarylogic_si256 (__m256i a, __m256i b, __m256i c, byte imm) VPTERNLOGD ymm1 {k1}{z}, ymm2, ymm3/m256, imm8 The above native signature does not exist. We provide this additional overload for consistency with the other bitwise APIs.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:12167]: Should some of this go in remarks? (same for next 2 changes)

File: xml/System.Runtime.Intrinsics.X86/Avx10v1.xml [hunk 20]
```diff
@@ -12354,7 +12354,7 @@
...
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx10v1.xml:12357]: Should some of this go in remarks? (same for next 2 changes)

File: xml/System.Runtime.Intrinsics.X86/Avx512F+X64.xml
Comment (gewarren) [inline xml/System.Runtime.Intrinsics.X86/Avx512F+X64.xml:1]: Second part of each summary (This intrinsic is only available...) should probably go in remarks.

File: xml/System.Runtime.Intrinsics/Vector128.xml [hunk 21]
```diff
@@ -4912,11 +4915,11 @@ On x86, this method corresponds to __m128i _mm_setr_epi8
...
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="start">To be added.</param>
-        <param name="step">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="T">The type of the elements in the vector.</typeparam>
+        <param name="start">The value that element 0 will be initialized to.</param>
+        <param name="step">The value that indicates how far apart each element should be from the previous.</param>
+        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector128`1" /> instance where the elements begin at a specified value and which are spaced apart according to another specified value.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector128.xml:4921]:
```suggestion
        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector128`1" /> instance where the elements begin at a specified value and are spaced according to another specified value.</summary>
```

File: xml/System.Runtime.Intrinsics/Vector256.xml [hunk 22]
```diff
@@ -4879,11 +4881,11 @@ On x86, this method corresponds to __m256i _mm256_setr_epi8
...
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="start">To be added.</param>
-        <param name="step">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="T">The type of the elements in the vector.</typeparam>
+        <param name="start">The value that element 0 will be initialized to.</param>
+        <param name="step">The value that indicates how far apart each element should be from the previous.</param>
+        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector256`1" /> instance where the elements begin at a specified value and which are spaced apart according to another specified value.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector256.xml:4887]:
```suggestion
        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector256`1" /> instance where the elements begin at a specified value and are spaced according to another specified value.</summary>
```

File: xml/System.Runtime.Intrinsics/Vector512.xml [hunk 23]
```diff
@@ -4411,11 +4414,11 @@
...
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="start">To be added.</param>
-        <param name="step">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="T">The type of the elements in the vector.</typeparam>
+        <param name="start">The value that element 0 will be initialized to.</param>
+        <param name="step">The value that indicates how far apart each element should be from the previous.</param>
+        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector512`1" /> instance where the elements begin at a specified value and which are spaced apart according to another specified value.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector512.xml:4420]:
```suggestion
        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector512`1" /> instance where the elements begin at a specified value and are spaced according to another specified value.</summary>
```

File: xml/System.Runtime.Intrinsics/Vector64.xml [hunk 24]
```diff
@@ -3775,11 +3775,11 @@ On x86, this method corresponds to __m64 _mm_setr_pi8
...
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="start">To be added.</param>
-        <param name="step">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <typeparam name="T">The type of the elements in the vector.</typeparam>
+        <param name="start">The value that element 0 will be initialized to.</param>
+        <param name="step">The value that indicates how far apart each element should be from the previous.</param>
+        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector64`1" /> instance where the elements begin at a specified value and which are spaced apart according to another specified value.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector64.xml:3781]:
```suggestion
        <summary>Creates a new <see cref="T:System.Runtime.Intrinsics.Vector64`1" /> instance where the elements begin at a specified value and are spaced according to another specified value.</summary>
```

File: xml/System.Runtime.Intrinsics/Vector64.xml [hunk 25]
```diff
@@ -6042,12 +6048,15 @@ This method may bypass the cache on certain platforms.
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="left">The vector to be multiplied with <paramref name="right" />.</param>
+        <param name="right">The vector to be multiplied with <paramref name="left" />.</param>
+        <param name="addend">The vector to be added to the result of <paramref name="left" /> multiplied by <paramref name="right" />.</param>
+        <summary>Computes an estimate of (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</summary>
+        <returns>An estimate of (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</returns>
+        <remarks>
+          <para>On hardware that natively supports <see cref="M:System.Runtime.Intrinsics.Vector64.FusedMultiplyAdd(System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double})" />, this may return a result that was rounded as one ternary operation.</para>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector64.xml:6057]:
```suggestion
          <para>On hardware that natively supports <see cref="M:System.Runtime.Intrinsics.Vector64.FusedMultiplyAdd(System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double})" />, this method might return a result that was rounded as one ternary operation.</para>
```

File: xml/System.Runtime.Intrinsics/Vector64.xml [hunk 26]
```diff
@@ -6042,12 +6048,15 @@ This method may bypass the cache on certain platforms.
...
+          <para>On hardware without specialized support, this may just return (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</para>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector64.xml:6058]:
```suggestion
          <para>On hardware without specialized support, this method might just return (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</para>
```

File: xml/System.Runtime.Intrinsics/Vector64.xml [hunk 27]
```diff
@@ -6071,12 +6080,15 @@ This method may bypass the cache on certain platforms.
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="left">The vector to be multiplied with <paramref name="right" />.</param>
+        <param name="right">The vector to be multiplied with <paramref name="left" />.</param>
+        <param name="addend">The vector to be added to the result of <paramref name="left" /> multiplied by <paramref name="right" />.</param>
+        <summary>Computes an estimate of (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</summary>
+        <returns>An estimate of (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</returns>
+        <remarks>
+          <para>On hardware that natively supports <see cref="M:System.Runtime.Intrinsics.Vector64.FusedMultiplyAdd(System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double})" />, this may return a result that was rounded as one ternary operation.</para>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector64.xml:6089]:
```suggestion
          <para>On hardware that natively supports <see cref="M:System.Runtime.Intrinsics.Vector64.FusedMultiplyAdd(System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double},System.Runtime.Intrinsics.Vector64{System.Double})" />, this method might return a result that was rounded as one ternary operation.</para>
```

File: xml/System.Runtime.Intrinsics/Vector64.xml [hunk 28]
```diff
@@ -6071,12 +6080,15 @@ This method may bypass the cache on certain platforms.
...
+          <para>On hardware without specialized support, this may just return (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</para>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector64.xml:6090]:
```suggestion
          <para>On hardware without specialized support, this method might just return (<paramref name="left" /> * <paramref name="right" />) + <paramref name="addend" />.</para>
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml (hunk 1 above)
Comment (tannergooding) [reply xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml:43]: I think there's a lot of missing newlines or `<para>` separators in these as well

File: xml/System.Runtime.Intrinsics/Vector128.xml [hunk 29]
```diff
@@ -1766,9 +1768,9 @@
         <Parameter Name="vector" Type="System.Runtime.Intrinsics.Vector128&lt;System.Single&gt;" Index="0" FrameworkAlternate="net-9.0" />
       </Parameters>
       <Docs>
-        <param name="vector">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="vector">The vector to convert.</param>
+        <summary>Converts a <see cref="T:System.Runtime.Intrinsics.Vector128`1" /> to a <see cref="T:System.Runtime.Intrinsics.Vector128`1" /> platform specific behavior on overflow.</summary>
```
Comment (gewarren) [inline xml/System.Runtime.Intrinsics/Vector128.xml:1772]:
```suggestion
        <summary>Converts a <see langword="Vector128{Single}" /> to a <see langword="Vector128{Int32}" /> using platform-specific behavior on overflow.</summary>
```

Comment (gewarren) [reply xml/System.Runtime.Intrinsics/Vector128.xml:1772]: The type parameters from https://github.com/dotnet/runtime/blob/6af853d7757e55a09fd9bc6dc1d6aa12c0b454f2/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs#L572 were lost here - and similarly throughout these APIs.

**Review by gewarren — CHANGES_REQUESTED** [review]

File: xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml:43]: @tannergooding I don't see the comments in triple slash split by `<para>`.

Comment (tannergooding) [reply xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml:43]: They're split by newline in the triple slash, which I don't think has been an issue with the other intrinsic docs; unless we manually fixed those all up in the docs?

File: xml/System.Runtime.Intrinsics/Vector128.xml (hunk 29 above)
Comment (carlossanlop) [reply xml/System.Runtime.Intrinsics/Vector128.xml]: I'll have to fix this later. Need to fix the bug in the tool first. Is that ok?

File: xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Runtime.Intrinsics.X86/Avx10v1+X64.xml]: I think there's a bug in the tool. I was expecting it to split new lines using `<para>`. I fixed it manually. Can you please take another look?

**Comment by carlossanlop** [comment]
@tannergooding @gewarren bad news: The compiler generated xmls do not preserve the type in generic types. For example:

https://github.com/dotnet/runtime/blob/50d6cad649aad2bfa4069268eddd16fd51ec5cf3/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs#L348-L353C44
```cs
        /// <summary>Converts a <see cref="Vector64{Single}" /> to a <see cref="Vector64{Int32}" /> using platform specific behavior on overflow.</summary>
        /// <param name="vector">The vector to convert.</param>
        /// <returns>The converted vector.</returns>
        [Intrinsic]
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static unsafe Vector64<int> ConvertToInt32Native(Vector64<float> vector)
```

When built, shows the following output under `D:\runtime\artifacts\bin\coreclr\windows.x64.Release\IL\System.Private.CoreLib.xml`:
```
        <member name="M:System.Runtime.Intrinsics.Vector128.ConvertToInt32Native(System.Runtime.Intrinsics.Vector128{System.Single})">
            <summary>Converts a <see cref="T:System.Runtime.Intrinsics.Vector128`1" /> to a <see cref="T:System.Runtime.Intrinsics.Vector128`1" /> platform specific behavior on overflow.</summary>
            <param name="vector">The vector to convert.</param>
            <returns>The converted vector.</returns>
        </member>
```

So our options are:
- Live with it.
- Fix them manually here (very labor intensive).
- Find a way to indicate the `<see crefs>` in triple slash comments that will preserve the typed generic argument. I suspect it might be possible, because as you can see, the DocId preserved the typed generic argument, but the comments did not.

**Comment by tannergooding** [comment]
> Find a way to indicate the <see crefs> in triple slash comments that will preserve the typed generic argument. I suspect it might be possible, because as you can see, the DocId preserved the typed generic argument, but the comments did not.

If there's a preferred way to state it instead of `<see cref="Vector64{Single}" />` then it should be possible to do a find/replace here to fix it up on the XML doc comment side.

Perhaps `<see cref="Vector64{T}" /> of <see cref="Single" />` or similar would be sufficient?

**Comment by tannergooding** [comment]
Would appreciate weigh-in from @gewarren on the preferred way for this to be "displayed" or otherwise surfaced for the docs page side of things, and then I can automate the replacement

For returns and similar, we do: [Vector128](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128-1?view=net-8.0)<[Int32](https://learn.microsoft.com/en-us/dotnet/api/system.int32?view=net-8.0)>

Where the first links to the `Vector128<T>`

So if there were maybe something like `<see cref="Vector64{T}" />&lt;<see cref="Single" />&gt;`, but with something to have the first part show as just `Vector64` and not as `Vector64<T>`

**Comment by gewarren** [comment]
> Would appreciate weigh-in from @gewarren on the preferred way for this to be "displayed" or otherwise surfaced for the docs page side of things, and then I can automate the replacement

My preference would just be use code tags or langword instead of `see cref`. E.g.:

```
Converts a <see langword="Vector64<Single>" /> to ...
```

**Comment by carlossanlop** [comment]
> and then I can automate the replacement

Thank you so much, @tannergooding. It's really frustrating that the intellisense xmls do not respect it when they're generated.

**Comment by gewarren** [comment]
Closing in favor of #10275.

**Pull request closed without merging**